tensornetwork/backends/numpy/decompositions.py

```diff
@@ -69,3 +69,40 @@ def svd_decomposition(
   vh = np.reshape(vh, [dim_s] + list(right_dims))
 
   return u, s, vh, s_rest
+
+def qr_decomposition(
```

Comment:
Can you add these two methods to the abstract `BaseBackend` as well? Just throw a `NotImplemented` error in both cases. This is so when/if we add a new backend (like with my jax_tpu backend https://github.com/google/TensorNetwork/pull/193), the user can know immediately that certain operations aren't supported.